lib_sw_dac/src/standard_fidelity/sigma_delta_1_5.S

```diff
@@ -27,22 +27,43 @@
     .linkset FUNCTION_NAME.nstackwords, 0
     .align 16
 FUNCTION_NAME:
-    { ldc r11, 0                ; nop }
-    { in r11, res[r0]            ; vsetc r11 }
+    // Init VPU
+    ldc r11, 0
+    vsetc r11
+// setup select
+    ldap r11, .sd_timeout; //r11 <- &sd_timeout handler
+    ldw r6, sp[SDAC_STACK_TIMEOUT_RESID];
+    setv res[r6], r11;  //set event vector
+    eeu res[r6]; // enable
+
+    ldap r11, .input_handler; //r11 <- input_handler
+    setv res[r0], r11;
+    eeu res[r0]; // enable
+
+    setsr 0x01; // event enable in thread
+
+    waiteu; // This does the initial IN or timeout
```

Comment:
I'd change this code to something like:
```
// Init VPU and the select statement
    { ldw r6, sp[SDAC_STACK_TIMEOUT_RESID]; ldc r11, 0  }
    { vsetc r11                 ; ldap r11, .sd_timeout }

    { setv res[r6], r11         ; ldap r11, .input_handler}
    eeu res[r6]; // enable

    setv res[r0], r11;
    eeu res[r0]; // enable

    waiteu; // This does the initial IN or timeout
```
Doesn't matter too much as it's just a startup, but it looks neater. Removed setsr as it's redundant, waiteu will enable events on the thread.